tests/unit/plugins/inventory/test_aws_ec2.py

```diff
@@ -277,43 +277,213 @@ def test_add_host_empty_hostnames(inventory):
+        }
+    ]
 
-    inventory._add_hosts(hosts, "aws_ec2", hostnames=["tag:Name", "private-dns-name", "dns-name"])
+    inventory._add_hosts(
+        hosts, "aws_ec2", hostnames=["tag:Name", "private-dns-name", "dns-name"]
+    )
     assert inventory.inventory.add_host.call_count == 1
     inventory.inventory.add_host.assert_called_with("sample-host", group="aws_ec2")
 
 
 def test_add_host_with_hostnames_and_two_matching_criteria(inventory):
-    hosts = [{
-        "Placement": {
-            "AvailabilityZone": "us-east-1a",
-        },
-        "PublicDnsName": "name-from-PublicDnsName",
-        "Tags": [{"Value": "name-from-tag-Name", "Key": "Name"}],
-    }]
+    hosts = [
+        {
+            "Placement": {
+                "AvailabilityZone": "us-east-1a",
+            },
+            "PublicDnsName": "name-from-PublicDnsName",
+            "Tags": [{"Value": "name-from-tag-Name", "Key": "Name"}],
+        }
+    ]
 
-    inventory._add_hosts(hosts, "aws_ec2", hostnames=["tag:Name", "private-dns-name", "dns-name"])
+    inventory._add_hosts(
+        hosts, "aws_ec2", hostnames=["tag:Name", "private-dns-name", "dns-name"]
+    )
     assert inventory.inventory.add_host.call_count == 1
-    inventory.inventory.add_host.assert_called_with("name-from-tag-Name", group="aws_ec2")
+    inventory.inventory.add_host.assert_called_with(
+        "name-from-tag-Name", group="aws_ec2"
+    )
 
 
-def test_add_host_with_hostnames_and_two_matching_criteria_and_allow_duplicated_hosts(inventory):
-    hosts = [{
-        "Placement": {
-            "AvailabilityZone": "us-east-1a",
-        },
-        "PublicDnsName": "name-from-PublicDnsName",
-        "Tags": [{"Value": "name-from-tag-Name", "Key": "Name"}],
-    }]
+def test_add_host_with_hostnames_and_two_matching_criteria_and_allow_duplicated_hosts(
+    inventory,
+):
+    hosts = [
+        {
+            "Placement": {
+                "AvailabilityZone": "us-east-1a",
+            },
+            "PublicDnsName": "name-from-PublicDnsName",
+            "Tags": [{"Value": "name-from-tag-Name", "Key": "Name"}],
+        }
+    ]
 
-    inventory._add_hosts(hosts, "aws_ec2", hostnames=["tag:Name", "private-dns-name", "dns-name"], allow_duplicated_hosts=True)
+    inventory._add_hosts(
+        hosts,
+        "aws_ec2",
+        hostnames=["tag:Name", "private-dns-name", "dns-name"],
+        allow_duplicated_hosts=True,
+    )
     assert inventory.inventory.add_host.call_count == 2
-    inventory.inventory.add_host.assert_any_call("name-from-PublicDnsName", group="aws_ec2")
+    inventory.inventory.add_host.assert_any_call(
+        "name-from-PublicDnsName", group="aws_ec2"
+    )
     inventory.inventory.add_host.assert_any_call("name-from-tag-Name", group="aws_ec2")
+
+
+def test_sanityize_hostname(inventory):
+    assert inventory._sanitize_hostname(1) == "1"
+    assert inventory._sanitize_hostname("a:b") == "a_b"
+    assert inventory._sanitize_hostname("a:/b") == "a__b"
```

Comment:
```suggestion
    assert inventory._sanitize_hostname("a:/b") == "a__b"
    assert inventory._sanitize_hostname("example") == "example"
```

I know it's the "I'm changing nothing" example, but I like to have one in the tests too since that's part of the contract of the function.